pkg/export/otel/metrics_svc_graph.go

```diff
@@ -0,0 +1,467 @@
+	m := mr.newSvcGraphMetricsInstance(service)
+
+	sglog().Debug("creating new metric set", "service", service)
+	// time units for HTTP and GRPC durations are in seconds, according to the OTEL specification:
+	// https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics/semantic_conventions
+	// TODO: set ExplicitBucketBoundaries here and in prometheus from the previous specification
+	meter := m.provider.Meter(reporterName)
+	var err error
+
+	if mr.cfg.ServiceGraphMetricsEnabled() {
+		err = mr.setupGraphMeters(&m, meter)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &m, nil
+}
+
+func (mr *SvcGraphMetricsReporter) close() {
+	if err := mr.exporter.Shutdown(mr.ctx); err != nil {
+		slog.With("component", "MetricsReporter").Error("closing metrics provider", "error", err)
+	}
+}
+
+func (mr *SvcGraphMetricsReporter) tracesResourceAttributes(service *svc.Attrs) attribute.Set {
+	if service == nil {
+		return *attribute.EmptySet()
+	}
+	baseAttrs := []attribute.KeyValue{
+		semconv.ServiceName(service.UID.Name),
+		semconv.ServiceInstanceID(service.UID.Instance),
+		semconv.ServiceNamespace(service.UID.Namespace),
+		semconv.TelemetrySDKLanguageKey.String(service.SDKLanguage.String()),
+		semconv.TelemetrySDKNameKey.String("opentelemetry-ebpf-instrumentation"),
+		request.SourceMetric(attr.VendorPrefix),
+		semconv.OSTypeKey.String("linux"),
+	}
+
+	extraAttrs := []attribute.KeyValue{
+		semconv.HostID(mr.hostID),
+	}
+
+	for k, v := range service.Metadata {
+		extraAttrs = append(extraAttrs, k.OTEL().String(v))
+	}
+
+	filteredAttrs := otelcfg.GetFilteredAttributesByPrefix(baseAttrs, nil, extraAttrs, MetricTypes)
+	return attribute.NewSet(filteredAttrs...)
+}
+
+func (mr *SvcGraphMetricsReporter) metricHostAttributes() attribute.Set {
+	attrs := []attribute.KeyValue{
+		GrafanaHostIDKey.String(mr.hostID),
+	}
+
+	return attribute.NewSet(attrs...)
+}
+
+// spanMetricAttributes follow a given specification, so their attribute getters are predefined and can't be
+// selected by the user
+func (mr *SvcGraphMetricsReporter) spanMetricAttributes() []attributes.Field[*request.Span, attribute.KeyValue] {
+	return append(attributes.OpenTelemetryGetters(
+		request.SpanOTELGetters, []attr.Name{
+			attr.ServiceName,
+			attr.ServiceInstanceID,
+			attr.ServiceNamespace,
+			attr.SpanKind,
+			attr.SpanName,
+			attr.StatusCode,
+			attr.Source,
+		}),
+		// hostID is not taken from the span but common to the metrics reporter,
+		// so the getter is injected here directly
+		attributes.Field[*request.Span, attribute.KeyValue]{
+			ExposedName: string(attr.HostID.OTEL()),
+			Get: func(_ *request.Span) attribute.KeyValue {
+				return semconv.HostID(mr.hostID)
+			},
+		})
+}
+
+func (mr *SvcGraphMetricsReporter) serviceGraphAttributes() []attributes.Field[*request.Span, attribute.KeyValue] {
+	return attributes.OpenTelemetryGetters(
```

Comment:
OK, so just to confirm, this PR doesn't actually add the k8s_cluster_name as an attribute to the service graph and it doesn't prefix the not default fields, as described [here](https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/connector/servicegraphconnector#section-readme), with `client_` and `server_`? I guess we need to follow-up with another PR on that after this is merged?

Reply:
You are right. This is just the refactor to allow a later, smaller PR fixing the issue. I preferred this way to avoid having an even bigger and disperse PR